### DISCUSSION TITLE
AWS SES 535 Error for invalid credentials

### DISCUSSION THREAD

**Discussion by rhadooqoo** [vercel/next.js#64695] [Help]
### Summary

I'm using Nodemailer to implement SMTP mail form AWS SES on my Next.js app. The problem is that when I deploy it on Vercel, and test the mail sending, gives me the error of 535 Invalid credentials. But if i run it on localhost it runs perfectly. I checked the environment variables and switched between SMTP Credentials and Access Keys over and over and still cannot figure it out. It used to work, rolled back the code, still not working. Thanks!

**Comment by ananuness** [comment]
Okay, there's a troubleshooting you can do based in your error:
- **Special characters in credentials:** so check your URLs and see if there's any of them or in any password
- **Incorrect SMTP login details:**  go to the navigation pane, select *SMTP Settings*. In the content pane, click on *Show User SMTP Credentials*. Thus, you will see the current SMTP credentials just to make sure there's no problem here

**Reply by rhadooqoo** [reply]
Thanks for the reply!
Problem is that AWS Ses always generates the password without any input from my part, and always has at least one special character in it... It used to work days before, but maybe then i had a password with no special characters...
